include/nbl/ext/FFT/FFT.h

```diff
@@ -0,0 +1,240 @@
+			pWrites[1].descriptorType = asset::EDT_STORAGE_BUFFER;
+			pWrites[1].count = 1;
+			pInfos[1].desc = outputBufferDescriptor;
+			pInfos[1].buffer.size = outputBufferDescriptor->getSize();
+			pInfos[1].buffer.offset = 0u;
+
+			driver->updateDescriptorSets(2u, pWrites, 0u, nullptr);
+		}
+
+		static inline void updateDescriptorSet(
+			video::IVideoDriver * driver,
+			video::IGPUDescriptorSet * set,
+			core::smart_refctd_ptr<video::IGPUImageView> inputImageDescriptor,
+			core::smart_refctd_ptr<video::IGPUBuffer> outputBufferDescriptor)
+		{
+			video::IGPUDescriptorSet::SDescriptorInfo pInfos[MAX_DESCRIPTOR_COUNT];
+			video::IGPUDescriptorSet::SWriteDescriptorSet pWrites[MAX_DESCRIPTOR_COUNT];
+
+			for (auto i=0; i< MAX_DESCRIPTOR_COUNT; i++)
+			{
+				pWrites[i].dstSet = set;
+				pWrites[i].arrayElement = 0u;
+				pWrites[i].count = 1u;
+				pWrites[i].info = pInfos+i;
+			}
+
+			// Input Buffer 
+			pWrites[0].binding = 0;
+			pWrites[0].descriptorType = asset::EDT_COMBINED_IMAGE_SAMPLER;
+			pWrites[0].count = 1;
+			pInfos[0].desc = inputImageDescriptor;
+			pInfos[0].image.sampler = nullptr;
+			pInfos[0].image.imageLayout = static_cast<asset::E_IMAGE_LAYOUT>(0u);;
+
+			// Output Buffer 
+			pWrites[1].binding = 1;
+			pWrites[1].descriptorType = asset::EDT_STORAGE_BUFFER;
+			pWrites[1].count = 1;
+			pInfos[1].desc = outputBufferDescriptor;
+			pInfos[1].buffer.size = outputBufferDescriptor->getSize();
+			pInfos[1].buffer.offset = 0u;
+
+			driver->updateDescriptorSets(2u, pWrites, 0u, nullptr);
+		}
+
+		static inline void dispatchHelper(
+			video::IVideoDriver* driver,
+			const DispatchInfo_t& dispatchInfo,
+			bool issueDefaultBarrier=true)
+		{
+			driver->dispatch(dispatchInfo.workGroupCount[0], dispatchInfo.workGroupCount[1], dispatchInfo.workGroupCount[2]);
+
+			if (issueDefaultBarrier)
+				defaultBarrier();
+		}
+
+		static inline void pushConstants(
+			video::IVideoDriver* driver,
+			video::IGPUPipelineLayout * pipelineLayout,
+			asset::VkExtent3D const & inputDimension,
+			asset::VkExtent3D const & paddedInputDimension,
+			Direction direction,
+			bool isInverse, 
+			PaddingType paddingType = PaddingType::CLAMP_TO_EDGE)
+		{
+			uint32_t is_inverse_u = isInverse;
+			driver->pushConstants(pipelineLayout, nbl::video::IGPUSpecializedShader::ESS_COMPUTE, 0u, sizeof(uint32_t) * 3, &inputDimension);
+			driver->pushConstants(pipelineLayout, nbl::video::IGPUSpecializedShader::ESS_COMPUTE, sizeof(uint32_t) * 4, sizeof(uint32_t) * 3, &paddedInputDimension);
+			driver->pushConstants(pipelineLayout, nbl::video::IGPUSpecializedShader::ESS_COMPUTE, sizeof(uint32_t) * 8, sizeof(uint32_t), &direction);
+			driver->pushConstants(pipelineLayout, nbl::video::IGPUSpecializedShader::ESS_COMPUTE, sizeof(uint32_t) * 9, sizeof(uint32_t), &is_inverse_u);
+			driver->pushConstants(pipelineLayout, nbl::video::IGPUSpecializedShader::ESS_COMPUTE, sizeof(uint32_t) * 10, sizeof(uint32_t), &paddingType);
+		}
```

Comment:
If you defined a `nbl_glsl_ext_FFT_Parameters_t` in the GLSL in its own little header (lets say `nbl/builtin/glsl/ext/FFT/parameters.glsl`), you could include it in this C++ header just so

```glsl
namespace impl
{

using uint = uint32_t;
struct uvec3
{
    uint x,y,z;
};
#include "nbl/builtin/glsl/ext/FFT/parameters.glsl"

}
struct Parameters_t : impl::nbl_glsl_ext_FFT_Parameters_t
{
    // add some methods for C++ only, but no members
};
```

Then you could use this struct throughout the codebase and use only one `pushConstants` call and not calculate your offsets and sizes manually